core/src/com/geekbrains/td/MonsterEmitter.java

```diff
@@ -1,12 +1,47 @@
 package com.geekbrains.td;
 
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.math.MathUtils;
+
+import java.util.LinkedList;
 
 public class MonsterEmitter extends ObjectPool<Monster> {
     private GameScreen gameScreen;
+    private float monsterTimer;
+    private LinkedList<MonsterWaveType> waveList;
+    private MonsterWaveType currentWave;
+    private int emittedMonsters;
+    private Monster singleMonster;
+    private boolean isSingleMonsterStatus;
+    private boolean isSingleFirstTime;
 
     public MonsterEmitter(GameScreen gameScreen) {
+        isSingleFirstTime = true;
+        isSingleMonsterStatus = false;
         this.gameScreen = gameScreen;
+        waveList = new LinkedList<MonsterWaveType>();
+        resetGeneration();
+    }
+
+    public boolean getSingleMonsterStatus() {
+        return isSingleMonsterStatus;
+    }
+
+    public void setSingleMonster(Monster singleMonster) {
+        this.singleMonster = singleMonster;
+    }
+
+    public LinkedList<MonsterWaveType> getWaveList() {
+        return waveList;
```

Comment:
You should never ever return the collection directly because you're returning actually a reference to it and it can be modified elsewhere by mistake. Consider it as a good coding style: always return `new ArrayList<>( waveList )` or `java.util.Collections.unmodifiableList( waveList )` and never return `waveList`.